pkg/ottl/e2e/e2e_test.go

```diff
@@ -84,12 +94,29 @@ func Test_e2e_editors(t *testing.T) {
 				m.PutStr("test.foo.flags", "pass")
 				m.PutStr("test.foo.slice.0", "val")
 				m.PutStr("test.foo.nested.test", "pass")
+
```

Comment:
@evan-bradley @TylerHelmuth - This is not related to the function implemented in this PR, but after adding a slice of nested objects to the test attributes I found that the `flatten` function seems to not flatten the attributes of nested objects within a slice, but rather leaves the objects within the slice unchanged - is this intended or is this a bug in that function?

Also, when setting the `depth` to `0` (as in the test case below), slices at the top level of the input of the flatten function will still be flattened.

Please let me know if this behavior is intended or if this may be a bug - If it's the latter, I can create an issue for that and work on a fix

Reply:
These sound like bugs.

Reply:
These sound like bugs to me, too. Could you file an issue?

Reply:
I have created two issues (I believe these are two distinct bugs):

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36161
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36162